app/config_test.go

```diff
@@ -49,3 +52,156 @@ func TestReadGroupDescToml(t *testing.T) {
+	require.Nil(t, err)
+
+	privateToml.WriteString(privateInfo)
+	privateToml.Close()
+
+	cothConfig, srv, err := ParseCothority(privateToml.Name())
+	require.Nil(t, err)
+
+	// Check basic information
+	require.Equal(t, suite, cothConfig.Suite)
+	require.Equal(t, public, cothConfig.Public)
+	require.Equal(t, private, cothConfig.Private)
+	require.Equal(t, address, cothConfig.Address.String())
+	require.Equal(t, listenAddr, cothConfig.ListenAddress)
+	require.Equal(t, description, cothConfig.Description)
+
+	srv.Close()
+}
+
+func TestParseCothorityWithTLSWebSocket(t *testing.T) {
+	suite := "Ed25519"
+	public := "6a921638a4ade8970ebcd9e371570f08d71a24987f90f12391b9f6c525be5be4"
+	private := "6a921638a4ade8970ebcd9e371570f08d71a24987f90f12391b9f6c525be5be4"
+	address := "tcp://1.2.3.4:1234"
+	listenAddr := "127.0.0.1:0"
+	description := "This is a description."
+
+	// Certificate and key examples taken from
+	// 'https://gist.github.com/blinksmith/579b2650a09f128a03ca'
+	wsTLSCert := `-----BEGIN CERTIFICATE-----
+MIICEzCCAXygAwIBAgIQMIMChMLGrR+QvmQvpwAU6zANBgkqhkiG9w0BAQsFADAS
+MRAwDgYDVQQKEwdBY21lIENvMCAXDTcwMDEwMTAwMDAwMFoYDzIwODQwMTI5MTYw
+MDAwWjASMRAwDgYDVQQKEwdBY21lIENvMIGfMA0GCSqGSIb3DQEBAQUAA4GNADCB
+iQKBgQDuLnQAI3mDgey3VBzWnB2L39JUU4txjeVE6myuDqkM/uGlfjb9SjY1bIw4
+iA5sBBZzHi3z0h1YV8QPuxEbi4nW91IJm2gsvvZhIrCHS3l6afab4pZBl2+XsDul
+rKBxKKtD1rGxlG4LjncdabFn9gvLZad2bSysqz/qTAUStTvqJQIDAQABo2gwZjAO
+BgNVHQ8BAf8EBAMCAqQwEwYDVR0lBAwwCgYIKwYBBQUHAwEwDwYDVR0TAQH/BAUw
+AwEB/zAuBgNVHREEJzAlggtleGFtcGxlLmNvbYcEfwAAAYcQAAAAAAAAAAAAAAAA
+AAAAATANBgkqhkiG9w0BAQsFAAOBgQCEcetwO59EWk7WiJsG4x8SY+UIAA+flUI9
+tyC4lNhbcF2Idq9greZwbYCqTTTr2XiRNSMLCOjKyI7ukPoPjo16ocHj+P3vZGfs
+h1fIw3cSS2OolhloGw/XM6RWPWtPAlGykKLciQrBru5NAPvCMsb/I1DAceTiotQM
+fblo6RBxUQ==
+-----END CERTIFICATE-----`
+	wsTLSCertKey := `-----BEGIN RSA PRIVATE KEY-----
+MIICXgIBAAKBgQDuLnQAI3mDgey3VBzWnB2L39JUU4txjeVE6myuDqkM/uGlfjb9
+SjY1bIw4iA5sBBZzHi3z0h1YV8QPuxEbi4nW91IJm2gsvvZhIrCHS3l6afab4pZB
+l2+XsDulrKBxKKtD1rGxlG4LjncdabFn9gvLZad2bSysqz/qTAUStTvqJQIDAQAB
+AoGAGRzwwir7XvBOAy5tM/uV6e+Zf6anZzus1s1Y1ClbjbE6HXbnWWF/wbZGOpet
+3Zm4vD6MXc7jpTLryzTQIvVdfQbRc6+MUVeLKwZatTXtdZrhu+Jk7hx0nTPy8Jcb
+uJqFk541aEw+mMogY/xEcfbWd6IOkp+4xqjlFLBEDytgbIECQQDvH/E6nk+hgN4H
+qzzVtxxr397vWrjrIgPbJpQvBsafG7b0dA4AFjwVbFLmQcj2PprIMmPcQrooz8vp
+jy4SHEg1AkEA/v13/5M47K9vCxmb8QeD/asydfsgS5TeuNi8DoUBEmiSJwma7FXY
+fFUtxuvL7XvjwjN5B30pNEbc6Iuyt7y4MQJBAIt21su4b3sjXNueLKH85Q+phy2U
+fQtuUE9txblTu14q3N7gHRZB4ZMhFYyDy8CKrN2cPg/Fvyt0Xlp/DoCzjA0CQQDU
+y2ptGsuSmgUtWj3NM9xuwYPm+Z/F84K6+ARYiZ6PYj013sovGKUFfYAqVXVlxtIX
+qyUBnu3X9ps8ZfjLZO7BAkEAlT4R5Yl6cGhaJQYZHOde3JEMhNRcVFMO8dJDaFeo
+f9Oeos0UUothgiDktdQHxdNEwLjQf7lJJBzV+5OtwswCWA==
+-----END RSA PRIVATE KEY-----`
+
+	// Write files containing cert and key
+	certFile, err := ioutil.TempFile("", "temp_cert.pem")
+	require.Nil(t, err)
+	certFile.WriteString(wsTLSCert)
+	certFile.Close()
+
+	keyFile, err := ioutil.TempFile("", "temp_key.pem")
```

Comment:
from godoc for TempFile: "It is the caller's responsibility to remove the file when no longer needed" See: https://godoc.org/io/ioutil#example-TempFile

Reply:
Ohh right! I was in fact looking at examples in our codebase and didn't see those removed, thus I assumed it was automatically done. For example, if you look in `platform_test.go` and search for `ioutil.TempFile`, are any of those removed in any way? Or should we add an issue to fix those?

Reply:
Yes, you can add an issue, thanks.